pkg/input/readdir.go

```diff
@@ -166,16 +203,23 @@ func ReadDir(fsys fs.FS, cfg config.Application, cfgFilePath string) (*core.Inpu
 			}
 			return nil
 		}
-		isPackageFile := false
+		isProjectFile := false
 		for _, lang := range allLangs {
-			if info.Name() == lang.packageFileName {
-				f, err = addFile(fsys, path, relPath, lang.packageFileOpener)
-				lang.foundPackageFile = true
-				isPackageFile = true
+			if lang.projectFilePredicate != nil && lang.projectFilePredicate(info.Name()) {
+				prjDir, _ := filepath.Split(info.Name())
+				if lang.foundProjectFile == true {
+					err = fmt.Errorf("multiple '%s' files found in directory: %s", lang.projectFileDescription, prjDir)
+					isProjectFile = true
+					break
+				}
+				f, err = addFile(fsys, path, relPath, lang.projectFileOpener)
+				lang.foundProjectFile = true
+				isProjectFile = true
+				projectDirs[lang.name] = append(projectDirs[lang.name], prjDir)
```

Comment:
is this also enabling the ability to have multiple languages in the same klotho compilation? Or what is projectDirs alluding to?

Reply:
not really. it's more just renaming references to "package" or "pkg" to "project" since package is more language-specific.

As far as I'm aware, there's nothing in `readdir` that limits the number of project files/languages we detect. The only logic that has limitations is `openFindUpward`, which is only executed we find 0 project files initially.

Reply:
`projectDirs` refers to directories in which we've detected project files for each supported language. For the moment it's used to enable language-specific file/directory exclusions, and to ensure that compilation fails if two project files exist for the same language in the same directory.

Support for multiple languages in a single compilation is not specifically handled here. The executable plugins don't currently support multiple languages in a single execution unit.